src/agent/src/config.rs

```diff
@@ -185,32 +213,32 @@ fn get_hotplug_timeout(param: &str) -> Result<time::Duration> {
     Ok(time::Duration::from_secs(value.unwrap()))
 }
 
-fn get_bool_value(param: &str, default: bool) -> bool {
+fn get_bool_value(param: &str) -> Result<bool> {
     let fields: Vec<&str> = param.split("=").collect();
 
     if fields.len() != 2 {
-        return default;
+        return Ok(false);
     }
 
     let v = fields[1];
 
     // bool
     let t: std::result::Result<bool, std::str::ParseBoolError> = v.parse();
     if t.is_ok() {
-        return t.unwrap();
+        return Ok(t.unwrap());
     }
 
     // integer
     let i: std::result::Result<u64, std::num::ParseIntError> = v.parse();
     if i.is_err() {
-        return default;
+        return Ok(false);
     }
 
     // only `0` returns false, otherwise returns true
-    match i.unwrap() {
+    Ok(match i.unwrap() {
         0 => false,
         _ => true,
-    }
+    })
```

Comment:
This `is_err()` `unwrap()` pattern can be replaced by a `if let` statement that is cleaner and less code.